### DISCUSSION TITLE
Question on configuration for lsp integration with Neovim

### DISCUSSION THREAD

**Discussion by nikfp** [lexical-lsp/lexical#511] [Q&A]
Hello all and thanks for the great tool! It's fast and doesn't overload my old machine which I really appreciate. 

I'm having a problem that I put my finger on the source of. Lexical is correctly reporting errors as I type, however once I correct the errors it's not refreshing the diagnostics being sent to Neovim's diagnostic API. So a missing `end` will continue to show a diagnostic warning even if I add the `end` for example. 

I integrated with Neovim following the instructions [here](https://github.com/lexical-lsp/lexical/blob/main/pages/installation.md#neovim)

I can't tell if this is configuration based on my part, or if this is an issue with the diagnostics reporting from Lexical. But if anyone has any insight I'm all ears.

**Comment by scohen** [comment]
I'm an emacs user, so I'm of no help here, though saving _should_ clear any open diagnostics.

@scottming Can you help?

**Comment by scottming** [comment]
@nikfp  Sorry, I missed this question.

If you have followed the neovim configuration I recommended, normally it shouldn't have this issue unless it's an heex file and you have also enabled functionalities like automatically formatting the file on save in the client.

If it's not a configuration issue, could you please provide some logs regarding this 'end' case? You should be able to find them in the `lexical.log` file.
